python/openassetio_manager_ftrack/FtrackInterface.py

```diff
@@ -0,0 +1,109 @@
+SETTINGS_KEY_API_KEY = "api_key"
+SETTINGS_KEY_API_USER = "api_user"
+
+
+class FtrackInterface(ManagerInterface):
+
+    __settings = {}
+    __session = None
+
+    def identifier(self):
+        return "com.ftrack"
+
+    def displayName(self):
+        return "ftrack"
+
+    def info(self):
+        # As we use a simple prefix, allow the OpenAssetIO C++ layer to
+        # short-circuit calls into Python for isEntityReferenceString
+        # where possible.
+        info = {constants.kField_EntityReferencesMatchPrefix: REFERENCE_PREFIX}
+        # Add any relevant server info if we have a session.
+        if self.__session:
+            if "version" in self.__session.server_information:
+                info["version"] = self.__session.server_information["version"]
+        return info
+
+    def settings(self, hostSession):
+        # Settings should always be a copy to avoid accidental updates
+        return self.__settings.copy()
+
+    def initialize(self, managerSettings, hostSession):
+        if self.__session:
+            self.__session.close()
+        self.__validate_settings(managerSettings)
+        self.__settings.update(managerSettings)
+        self.__session = self.__create_session(self.__settings, hostSession)
+
+    def managementPolicy(self, traitSets, context, hostSession):
+        # Ignore everything for now
+        return [TraitsData() for _ in traitSets]
+
+    def isEntityReferenceString(self, someString, hostSession):
+        return someString.startswith(self.__reference_prefix)
+
+    @staticmethod
+    def __validate_settings(managerSettings):
+        """
+        Ensures settings only contain valid keys, raising if not.
+        (required by OpenAssetIO API contract)
+        """
+        valid_keys = {SETTINGS_KEY_SERVER, SETTINGS_KEY_API_USER, SETTINGS_KEY_API_KEY}
+        for key, value in managerSettings.items():
+            if key not in valid_keys:
+                raise KeyError(f"Unrecognised setting '{key}'")
+            if not isinstance(value, str):
+                raise ValueError(f"Invalid type for setting '{key}' must be a string")
+
+    @staticmethod
+    def __create_session(settings, hostSession):
+        """
+        Creates and returns a new ftrack API session, using
+        configuration from the environment if specific values are
+        missing from the supplied settings.
+        """
+        logger = hostSession.logger()
+
+        # We defer this to avoid any heavy lifting when
+        # the interface is constructed, but not yet
+        # initialized (e.g. to present in a UI selector).
+        # pylint: disable=import-outside-toplevel
+        import ftrack_api
+
+        server_url = settings.get(SETTINGS_KEY_SERVER, os.environ.get("FTRACK_SERVER"))
+        api_key = settings.get(SETTINGS_KEY_API_KEY, os.environ.get("FTRACK_API_KEY"))
+        api_user = settings.get(
+            SETTINGS_KEY_API_USER, os.environ.get("FTRACK_API_USER")
```

Comment:
With USER I was trying to avoid validating the input in the manager plugin due to [this note](http://ftrack-python-api.rtd.ftrack.com/en/stable/environment_variables.html#envvar-FTRACK_API_USER) on the behaviour when it was missing, which some people might be making use of. But the messages from the SDK aren't as helpful as one would like.

Reply:
OK, reasonable point on not validating too early.  We could catch the exception and prefix it with a nicer error?  But happy to leave that for now, it was only a `[minor]` - people _should_ read the instructions :grin: .